apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -81,14 +84,46 @@ class LibraryCreationDialog extends React.Component {
       this.state.libraryClientApi,
       error =>
         this.setState({dialogState: DialogState.ERROR, errorMessage: error}),
-      libraryDetails =>
-        this.setState({
-          dialogState: DialogState.DONE_LOADING,
-          libraryDetails: libraryDetails
-        })
+      this.onLibraryLoaded
     );
   };
 
+  onLibraryLoaded = async libraryDetails => {
+    const defaultNewState = {
+      dialogState: DialogState.DONE_LOADING,
+      libraryDetails
+    };
+
+    try {
+      const profaneWords = await this.findProfanity(
```

Comment:
Why do we need to do this here and in the `LibraryPublisher`?

Reply:
here, we check for profanity in the project code (if there's profanity we don't allow them to even attempt to publish a library). then in `LibraryPublisher` we check the user input (library name and description)